supabase/migrations/20240223012557_sentences.sql

```diff
@@ -0,0 +1,182 @@
+CREATE TABLE sentences (
+  id uuid unique primary key NOT NULL, -- generated on client so users can create a sentence offline and keep editing it
+  "text" text,
+  translation jsonb,
+  created_at timestamp with time zone DEFAULT now() NOT NULL,
+  created_by uuid NOT NULL REFERENCES auth.users (id),
+  updated_at timestamp with time zone DEFAULT now() NOT NULL,
+  updated_by uuid NOT NULL REFERENCES auth.users (id),
+  deleted timestamp with time zone
+);
+
+ALTER TABLE sentences ENABLE ROW LEVEL SECURITY;
+
+CREATE TABLE senses_in_sentences (
+  sense_id uuid NOT NULL REFERENCES senses (id),
+  sentence_id uuid NOT NULL REFERENCES sentences (id),
+  created_by uuid NOT NULL REFERENCES auth.users (id),
+  created_at timestamp with time zone NOT NULL DEFAULT now(),
+  deleted timestamp with time zone,
+  PRIMARY KEY (sense_id, sentence_id)
+);
+
+ALTER TABLE senses_in_sentences ENABLE ROW LEVEL SECURITY;
+
+CREATE TYPE sentence_tables AS ENUM ('sentences', 'senses_in_sentences');
+CREATE TYPE sentence_columns AS ENUM ('deleted', 'text', 'translation');
+
+CREATE TABLE sentence_updates (
+  id uuid unique primary key NOT NULL, -- generated by client via uuidv4 so it can be idempotent and they can send it multiple times without repeated effect in case of network issues
+  user_id uuid NOT NULL REFERENCES auth.users (id),
+  firebase_email text, -- temporary until migration complete
+  dictionary_id text NOT NULL REFERENCES dictionaries (id),
+  sentence_id uuid NOT NULL REFERENCES sentences (id),
+  sense_id uuid REFERENCES senses (id), -- could be null, but if included the apply_sentence_updates trigger will create a sense-sentence relationship
+  "timestamp" timestamp with time zone NOT NULL DEFAULT now(),
+  "table" sentence_tables NOT NULL,
+  -- all following could be null if just adding a new sense-sentence relationship using senses_in_sentences table
+  "column" sentence_columns, 
+  new_value text, 
+  old_value text
+);
+
+ALTER TABLE sentence_updates ENABLE ROW LEVEL SECURITY;
+
+CREATE OR REPLACE FUNCTION ensure_dictionary_before_apply_sentence_updates()
+RETURNS TRIGGER AS $$
+DECLARE
+  column_data_type regtype;
```

Comment:
@jacob-8 Why are you declaring these variables if you're not using them here? 
Are they for future changes?